doc/source/operations/rabbitmq.rst

```diff
@@ -0,0 +1,103 @@
+  Pulp.
+* Enable the high availability setting via Kolla-Ansible
+  ``om_enable_rabbitmq_high_availability``.
+
+By default in Kolla-Ansible, two key options for the high availability of
+RabbitMQ are disabled. These are durable queues, where messages are persisted
+to disk; and classic queue mirroring, where messages are replicated across
+multiple exchanges. Without these, a deployment will be a poor state for
+managing any updates for RabbitMQ, or any outages that cause it to be stopped.
+Messages held in RabbitMQ nodes that are stopped will be lost, which causes
+knock-on effects to the OpenStack services which either sent or were expecting
+to receive them. The Kolla-Ansible flag
+``om_enable_rabbitmq_high_availability`` can be used to enable both of these
+options. This will default to ``true`` from Xena onwards.
+
+While the `RabbitMQ docs <https://www.rabbitmq.com/queues.html#durability>`_ do
+say "throughput and latency of a queue is not affected by whether a queue is
+durable or not in most cases", it should be mentioned that there could be a
+potential performance hit from replicating all messages to the disk within
+large deployments. These changes would therefore be a tradeoff of performance
+for stability.
+
+**NOTE:** There is guaranteed to be downtime during this procedure, as it
+requires restarting RabbitMQ and all the OpenStack services that use it. The
+state of RabbitMQ will also be reset.
+
+Instructions
+------------
+
+If you are upgrading from Wallaby to Xena, or if you're on Wallaby and want
+RabbitMQ to be more stable, you will need to enable the HA config option in
+``etc/kayobe/kolla/globals.yml``.
+
+.. code-block:: console
+
+  om_enable_rabbitmq_high_availability: true
+
+Synchronise the Pulp containers.
+
+.. code-block:: console
+
+  kayobe playbook run etc/kayobe/ansible/pulp-container-sync.yml pulp-container-publish.yml -e stackhpc_pulp_images_kolla_filter=rabbitmq
+
+Generate the new config files for the overcloud services.
+
+.. code-block:: console
+
+  kayobe overcloud service configuration generate
+
+Pull the RabbitMQ container image.
+
+.. code-block:: console
+
+  kayobe overcloud container image pull -kt rabbitmq
+
+Stop all the OpenStack services which use RabbitMQ.
+
+.. code-block:: console
+
+  kayobe overcloud host command run --command "docker ps -a | egrep '(barbican|blazar|ceilometer|cinder|cloudkitty|designate|heat|ironic|keystone|magnum|manila|masakari|neutron|nova|octavia)' | awk '{ print $NF }' | xargs docker stop"
+
+Upgrade RabbitMQ.
+
+.. code-block:: console
+
+  kayobe overcloud service upgrade -kt rabbitmq --skip-prechecks
+
+In order to convert the queues to be durable, you will need to reset the state
+of RabbitMQ, and restart the services which use it. This can be done with the
+RabbitMQ hammer playbook:
+
+.. code-block:: console
+
+  kayobe playbook run stackhpc-kayobe-config/etc/kayobe/ansible/rabbitmq-reset.yml
```

Comment:
Does this playbook restart all of the services we stopped?

Reply:
Oh, no it just restarts the services that have been seen to break when RabbitMQ goes down. IMO it's best to keep the playbook only targeting this subset so it can be used separately if rabbit breaks. So I'll add the step to start all the services back into the instructions again.